src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -278,4 +285,38 @@ private boolean isConnectedToPedestrianNavigableHighway(final Edge edge)
     {
         return edge.connectedEdges().stream().anyMatch(HighwayTag::isPedestrianNavigableHighway);
     }
+
+    /**
+     * Checks if the edge is within or intersects airport polygon. Airport polygon is an atlas area
+     * with {@link AerowayTag} tag.
+     *
+     * @param edge
+     *            any edge
+     * @return true if the edge is within or intersects airport polygon
+     */
+    private boolean intersectsAirportOrBuilding(final Edge edge)
+    {
+        return StreamSupport.stream(edge.getAtlas()
+                .areasIntersecting(edge.bounds(),
```

Comment:
With using the `edge.bounds()` for the intersect there is a possibility of false negatives just outside airports, right? Albeit, the chance is probably quite low.

Reply:
I would say the probability would be very low. One other way would be to grab all areas containing the nodes of the edge matching a predicate. What do you think?

Reply:
Check this out for a solution I used a long while ago:
https://github.com/osmlab/atlas-checks/blob/dev/src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidAccessTagCheck.java#L112

However, in this case it may be so unlikely that it is not worth the extra processing.

Reply:
To add to this--just to be safe we could check if the airport intersects with the edge itself, not the edge's bounds. This could be accomplished using the ```overlaps()``` function of ```GeometricSurface```. Here's the method description for ```overlaps()``` in ```Polygon```:
```
     * Tests if this {@link Polygon} intersects/overlaps the given {@link PolyLine} at any point.
     * This is different than the {@link #fullyGeometricallyEncloses(PolyLine)} method, in that it
     * doesn't require full containment within the {@link Polygon}, just any overlap.
```
We could do something like ```AirportPolygon.overlaps(Edge)```. Internally it looks like this is comparing lat/longs of the concerned objects.